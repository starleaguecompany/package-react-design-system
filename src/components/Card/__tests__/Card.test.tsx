import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Card } from '../index';

describe('Card', () => {
  test('should render correctly with no props', () => {
    render(<Card />);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('container size-24 color-light-primary', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Card id="test-id" className="test-class" />);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('test-class');
    expect(screen.getByTestId('Card')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with content', () => {
    render(<Card>Test content</Card>);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByText('Test content')).toBeVisible();
  });

  test('should render correctly with shadow prop', () => {
    render(<Card shadow>Testing card</Card>);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('shadow');
  });

  test('should render correctly with variant prop', () => {
    render(<Card variant="primary">Testing card</Card>);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('color-light-primary');
  });

  test('should render correctly with color prop', () => {
    render(<Card color="blue">Testing card</Card>);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('color-blue-primary');
  });

  test('should render correctly with color and variant props', () => {
    render(
      <Card color="blue" variant="outlined">
        Testing card
      </Card>
    );

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('color-blue-outlined');
  });

  test('should render correctly with size prop', () => {
    render(<Card size={40}>Testing card</Card>);

    expect(screen.getByTestId('Card')).toMatchSnapshot();
    expect(screen.getByTestId('Card')).toHaveClass('size-40');
  });
});
