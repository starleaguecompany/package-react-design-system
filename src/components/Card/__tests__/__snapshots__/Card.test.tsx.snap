// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly with attributes 1`] = `
<div
  class="test-class container size-24 color-light-primary"
  data-qa="Card"
  id="test-id"
/>
`;

exports[`Card should render correctly with color and variant props 1`] = `
<div
  class="container size-24 color-blue-outlined"
  data-qa="Card"
>
  Testing card
</div>
`;

exports[`Card should render correctly with color prop 1`] = `
<div
  class="container size-24 color-blue-primary"
  data-qa="Card"
>
  Testing card
</div>
`;

exports[`Card should render correctly with content 1`] = `
<div
  class="container size-24 color-light-primary"
  data-qa="Card"
>
  Test content
</div>
`;

exports[`Card should render correctly with no props 1`] = `
<div
  class="container size-24 color-light-primary"
  data-qa="Card"
/>
`;

exports[`Card should render correctly with shadow prop 1`] = `
<div
  class="container size-24 color-light-primary shadow"
  data-qa="Card"
>
  Testing card
</div>
`;

exports[`Card should render correctly with size prop 1`] = `
<div
  class="container size-40 color-light-primary"
  data-qa="Card"
>
  Testing card
</div>
`;

exports[`Card should render correctly with variant prop 1`] = `
<div
  class="container size-24 color-light-primary"
  data-qa="Card"
>
  Testing card
</div>
`;
